pkg/node/node.go

```diff
@@ -488,6 +494,27 @@ func (s *CSINodeService) NodePublishVolume(ctx context.Context, req *csi.NodePub
 			resp, errToReturn = nil, fmt.Errorf("failed to publish volume: fake attach error %s", err.Error())
 		}
 	} else {
+		// will check whether srcPath is mounted, if not, need to redo NodeStageVolume
```

Comment:
I am a little concerned about this process,  as some mount logic(like global path setup) is done in kubelet.  there redo nodeStage may not help in such case.  do we have a test can prove that this will help in the failure case?

Reply:
The kubelet's wrong CSI call NodePublishVolume in this case assumed that the volume's device global path's mountpoint has been successfully setup in some successful previous CSI call NodeStageVolume. But acutally the volume's device global path has been unmounted in the forceful-node-removal (possibly) and kubelet cannot successfully sync the volume's real status because of its continually-failed "orphan" pod's volume cleanup by CSI call NodeUnpublish. 

In spite of this, the device global path still exists there if the volume still exists in the most cases. Even though the device global path has also been removed in the worst case, CSI current logic can still create the device global path itself.

I will do the functional test on this code change to verify it.

Reply:
I've similuated the scenario of volume's k8s global device mountpoint missing in my standalone test. This CSI defensive enhancement can work well as expected in the test.